### ISSUE TITLE
Add Wherigo cartridge hosting

### ISSUE THREAD

**Issue by andrixnet** [opencaching/opencaching-pl#591]
OPENCACHING-PL site offers file hosting (with certain limitations) for images and MP3 audio files for caches.
There is also an attribute "Wherigo" for GPS adventure based on the Wherigo platform.
This requires that the cache owner provides the wherigo cartridge file for players to download.
Currently there is no option other then third party hosting.

This is a feature request that wherigo cartridges be uploaded and hosted like images and MP3 files and attached to the cache.

**Comment by kojoty** [comment]
I agree - wherigo support could be a nice feature (wherigo attribute or maybe even a type of cache etc.)

@andrixnet, can you know if we can implement wherigo support - I mean if there are not any licence limitations or something? (for example maybe the "wherigo" keyword is a protected trademark?)

**Comment by andrixnet** [comment]
I would suggest several new cache types, including Wherigo, rather then attribute. IMO this makes OC sites richer. 

"wherigo.com" is listed as "Groundspeak Project". http://www.wherigo.com/ see bottom of page. It shows the term as "WherigoTM"
Groundspeak terms of use list the keyword "wherigo" in 2 places, however neither claiming trademark status.
According to Wikipedia, "wherigo platform"  was created by Groundspeak. https://en.wikipedia.org/wiki/Wherigo

There is a claim on a different site, apparently related, but not directly affiliated with Groundspeak : http://www.wherigofoundation.com/

> Wherigo is a registered trademark of Groundspeak Inc. The Wherigo design, platform, and associated intellectual property are also owned
> by Groundspeak Inc. They are used herein with Provisional Permission. Copyright 2015. All Rights Reserved.

**However, this foundation declares: **

> The Wherigo Foundation is an organized group of enthusiasts who have come together to develop Wherigo beyond Groundspeak’s initial specification. This volunteer group is comprised of Wherigo Builder and Player developers, artists, and web developers. The group’s immediate goal is to make Wherigo more accessible and user friendly to the community while working on the long-term goal of reimagining the Wherigo experience.
> 
> The foundation is responsible for everything regarding Wherigo: the builders and compilers that make, emulators that test, site that hosts, and apps that play the cartridges. If you would like to volunteer your time, please contact the person in charge of the project to which you would like to contribute.

See their guidelines as well : http://www.wherigofoundation.com/help/guidelines.aspx 

Wherigo builder by Groundspeak was last released in 2008 in alpha stage, is buggy and works poorly. Platform support stops at some Garmin GPS devices and Windows CE (Pocket PC)

There is a new editor by a czech team: https://www.urwigo.cz/podpora
I have contacted them regarding such legal matters.

There is also WhereYouGo (android app) https://play.google.com/store/apps/details?id=menion.android.whereyougo&hl=en 
I have contacted them regarding such legal matters.

Waiting for reply from the above. :-)

**Comment by andrixnet** [comment]
@kojoty how many caches on OCPL have "wherigo" attribute set? 
@harrieklomp how many caches on OCNL have "wherigo" attribute set? 

If it requires legal clearance, then it should be obtained (I doubt it is obtainable, OC since is direct competition to Groundspeak, even though wherigo is likely abandonware since 2009?) or the feature dropped entirely? 

(if dropped and user chooses to publish with cartridge, using external reference to download it, then rules should either forbid it, or allow it on the user's own responsability, but should be clarified explicitely)

**Comment by kojoty** [comment]
> how many caches on OCPL have "wherigo" attribute set? 

OCPL doesn't have "wherigo" attribute and I don't know about any such cache at ocpl (I pass one "secret-agent-wherigo-mission" from GC and it was quite nice).

**Comment by deg-pl** [comment]
But... we have ;)
@kojoty look at http://opencaching.pl/viewcache.php?cacheid=25367 for example

**Comment by kojoty** [comment]
thanks @deg - update: I was wrong **we have such attribute and such caches :)**

**Comment by andrixnet** [comment]
Is there any chance of advancing in this regard ? What can I help with? (_at this time I think I can make at least the wherigo attribute icon :-) _)

About usage, there is also the URWIGO project, the czech guys who made an alternate builder https://www.urwigo.cz/ 
The site is mostly czech, but maybe we could talk to them about the matter?

**Comment by andrixnet** [comment]
I know there is the attribute. That is beside the point. 

Of course they put link in description, because they have no direct option (unlike the MP3 podcache). So likely dually published cache anyway.

www.wherigo.com is a de-facto abandonware platform. It still works, but it hasn't received updates since like 2008-2009 with only a few links to changed Groundspeak docs and a cookie selector ...
Their builder is ALPHA stage, last updated in 2008.
They list "_Pocket PC_" and "_Garmin Oregon_" as featured tools.
Their compile service provides "_Pocket PC_" or "_Garmin Colorado_" as target device options... 
Such devices no longer exist except in the collection of an old-tech enthuziast. 

I've had repeated experiences trying to publish wherigo caches on geocaching.com (_following their rule that the cartridge has to be uploaded to wherigo.com first_) with wherigo.com failing to compile and accept the cartridge (build with Urwigo). 
Some I managed to get published, having the cartridge hosted separately elsewhere (fortunately this was relatively easy for me).

**Another reason**: why make OC users depend on thirt party service and a mandatory GC account to publish a wherigo cartridge on OPENCACHING? 

Thank you.
